### ISSUE TITLE
split modules for improved performance

### ISSUE THREAD

**Issue by balmas** [alpheios-project/node-build#16]
I think this represents decisions from slack and check-in discussions:

1) switch to using rollup to build ecma6 modules
2) bundle all 3rd party dependencies in one package
3) build individual bundles for each Alpheios repository
4) user a master file and dynamic imports to simplify include for embed-lib and webextension 

@kirlat does that sound right?

**Comment by kirlat** [comment]
Yes, absolutely. I would only say that for (2) we might explore having either all vendor libs in one bundle or having a bundle for each third party lib (big enough, of course so we wont' have 10+ vendor bundles) on a condition it will not complicate the use of the embedded lib to the user (i.e. those separate bundles will be invisible to the end user as they be loaded internally). Advantage of having it bundled separately is that we won't need to redistribute the whole bundle every time any third-party library is updated. It will be enough to redistribute just the updated library. Would it be useful to us, what do you think?

**Comment by balmas** [comment]
yes, I think that works, especially if we can do (4) i.e. have a master file managing dynamic imports of everything for simplicity of use

**Comment by balmas** [comment]
dynamic imports tested on:

@monzug : Chrome and firefox on desktop, Samsung 5s, iPhone xr (safari + Chrome) 
@balmas: Chrome on Linux, Safari on Mac, Chrome on Samsung Galaxy 7
@kirlat:  
  Win10 in Chrome and Firefox 
  MacBook Air in Chrome and Safari
  Redmi 1S in Chrome
  iPhone 6S in Chrome and Safari
@irina060981: 
  desktop (Chrome,  Firefox)
  iPhone 7 Plus (Safari)
 Samsung A5(2016)  (Chrome)

**Comment by balmas** [comment]
taking this out of the release. we have done some preparation for this by switching to a dynamic import of the components library and including all dependencies in library.
